sage-android/Sage/app/src/main/java/blueprint/com/sage/announcements/EditAnnouncementFragment.java

```diff
@@ -0,0 +1,188 @@
+            mAnnouncementSchoolList.setVisibility(View.VISIBLE);
+            School school = mAnnouncement.getSchool();
+            for (int i = 0; i < mSchools.size(); i++) {
+                if (mSchools.get(i).getId() == school.getId()) {
+                    mAnnouncementSchoolList.setSelection(i);
+                }
+            }
+        } else {
+            mAnnouncementCategory.setSelection(0);
+        }
+        setSpinnerListeners();
+    }
+
+    private boolean isValid() {
+        return mValidator.hasNonBlankField(mAnnouncementTitle, "Title") &
+                mValidator.hasNonBlankField(mAnnouncementBody, "Body");
+    }
+
+    public void makeRequest() {
+        if (!isValid())
+            return;
+        mAnnouncement.setTitle(mAnnouncementTitle.getText().toString());
+        mAnnouncement.setBody(mAnnouncementBody.getText().toString());
+        Requests.Announcements.with(getActivity()).makeEditRequest(mAnnouncement);
+
+    }
+
+    public void setSpinnerListeners() {
+        mAnnouncementCategory.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                if (mAnnouncementCategory.getSelectedItem().equals("General Announcement")) {
+                    mAnnouncement.setCategory(1);
+                    mAnnouncement.setSchoolId(0);
+                    mAnnouncement.setSchoolName(null);
+                    mAnnouncementSchoolList.setVisibility(View.GONE);
+                } else if (mAnnouncementCategory.getSelectedItem().equals("School Announcement")) {
+                    mAnnouncement.setCategory(0);
+                    mAnnouncementSchoolList.setVisibility(View.VISIBLE);
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+            }
+        });
+        mAnnouncementSchoolList.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                School school = (School) adapterView.getSelectedItem();
+                mAnnouncement.setSchoolId(school.getId());
+                mAnnouncement.setSchoolName(school.getName());
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+            }
+        });
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        EventBus.getDefault().unregister(this);
+    }
+
+    public void onEvent(EditAnnouncementEvent event) {
+        FragUtils.popBackStack(this);
+    }
+
+    public void onEvent(SchoolListEvent event) {
+        mSchools = event.getSchools();
```

Comment:
you should change the school setting thing for your spinner as a separate method, because for editing fragment, once you get the list of schools, you should set the school of the spinner to the announcement's current school (if applicable)